crates/node/p2p/src/net/broadcast.rs

```diff
@@ -34,17 +39,40 @@ impl Broadcast {
     /// If an error occurs, the function will exit early, maintaining the invariant that unsafe
     /// blocks are held in order.
     pub fn broadcast(&mut self) {
-        loop {
-            if self.buffer.is_empty() {
-                break;
-            }
-            if let Some(block) = self.buffer.front() {
-                if let Err(e) = self.channel.send(block.clone()) {
-                    trace!("Failed to send unsafe block through channel: {:?}", e);
+        if self.broadcasting.lock().map_or(true, |v| *v) {
+            trace!(target: "net", "Broadcasting in flight, skipping");
+            return;
+        }
+        if let Ok(mut broadcasting) = self.broadcasting.lock() {
+            *broadcasting = true;
+        }
+        let flag = Arc::clone(&self.broadcasting);
+        let channel = self.channel.clone();
+        let mut drained = self.buffer.drain(..).collect::<VecDeque<_>>();
+        tokio::spawn(async move {
+            let mut count = 0;
+            const MAX_THRESHOLD: usize = 5;
+            loop {
+                if drained.is_empty() {
+                    break;
+                }
+                if count > MAX_THRESHOLD {
+                    warn!(target: "net", threshold = MAX_THRESHOLD, "Broadcast attempts exceeded max threshold");
                     break;
                 }
-                self.buffer.pop_front();
+                if let Some(block) = drained.front() {
+                    if let Err(e) = channel.send(block.clone()) {
+                        trace!(target: "net", ?e, "Failed to broadcast block");
+                        count += 1;
+                    } else {
+                        count = 0;
+                    }
+                    drained.pop_front();
+                }
             }
```

Comment:
We can use `backon` here instead of a hand-rolled backoff strategy; Once this is rebased on #1753 should be able to introduce it.